internal/util/telemetry/reporter.go

```diff
@@ -78,11 +78,17 @@ type NewReporterOpts struct {
 
 // NewReporter creates a new reporter.
 func NewReporter(opts *NewReporterOpts) (*Reporter, error) {
-	t, err := initialState(opts.F, opts.DNT, opts.ExecName, opts.P.Get().Telemetry, opts.L)
+	t, lock, err := initialState(opts.F, opts.DNT, opts.ExecName, opts.P.Get().Telemetry, opts.L)
 	if err != nil {
 		return nil, err
 	}
 
+	if t != nil && lock {
+		if err = opts.P.Update(func(s *state.State) { s.TelemetryLocked = true }); err != nil {
+			return nil, err
+		}
```

Comment:
Those lines should be covered by test

Reply:
https://github.com/FerretDB/FerretDB/pull/1380/commits/3d6a9d7bc1ee0fd19e0fa7fcead28f7334d87157 done